server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/cluster/ProcessInfoProvider.java

```diff
@@ -25,38 +25,37 @@
 import org.sonar.api.Startable;
 import org.sonar.api.ce.ComputeEngineSide;
 import org.sonar.api.server.ServerSide;
-import org.sonar.process.cluster.info.ProcessInfoProvider;
-import org.sonar.process.cluster.info.ProcessInfoProviderRef;
 import org.sonar.process.systeminfo.Global;
 import org.sonar.process.systeminfo.SystemInfoSection;
 import org.sonar.process.systeminfo.protobuf.ProtobufSystemInfo;
 
 @ServerSide
 @ComputeEngineSide
-public class ProcessInfoProviderImpl implements ProcessInfoProvider, Startable {
+public class ProcessInfoProvider implements Startable {
 
+  /** */
+  private static ProcessInfoProvider INSTANCE;
   private final List<SystemInfoSection> sections;
 
-  public ProcessInfoProviderImpl(SystemInfoSection[] sections) {
+  public ProcessInfoProvider(SystemInfoSection[] sections) {
     this.sections = Arrays.stream(sections)
       .filter(section -> !(section instanceof Global))
       .collect(Collectors.toList());
   }
 
   @Override
   public void start() {
-    ProcessInfoProviderRef.setProcessInfoProvider(this);
+    INSTANCE = this;
   }
 
   @Override
   public void stop() {
-    ProcessInfoProviderRef.setProcessInfoProvider(null);
+    INSTANCE = null;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Synchronize this lazy initialization of 'INSTANCE' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)